lib/index.js

```diff
@@ -151,13 +151,107 @@ Appboy.prototype.initializeV2 = function(customEndpoint) {
   window.appboy.initialize(options.apiKey, config);
   
   if (options.automaticallyDisplayMessages) window.appboy.display.automaticallyShowNewInAppMessages();
-  if (userId) window.appboy.changeUser(userId);
-  
-  window.appboy.openSession();
+  this.load('v2', function() {
+    console.log("IBM_TEST: ready and changing user", userId);
+    var userId = self.analytics.user().id();
+
+    if (userId) {
+      window.appboy.changeUser(userId, self._createMessagingReadyCallback(function() {
```

Comment:
Could we store the return value of `self._createMessagingReadyCallback()` in a variable, then pass in `openSession` with that value (the global callback) as the `changeUser` `messagingReadyCallback`?

e.g.
```
var globalMessagingReadyCallback = self._createMessagingReadyCallback();
if (userId) {
  window.appboy.changeUser(userId, function() { 
    window.openSession(globalMessagingReadyCallback); 
  });
else {
  window.appboy.openSession(globalMessagingReadyCallback);
}
```

I think that would allow waiting for `openSession` to finish as well in case that makes a trigger fetch (I believe that would happen if we've cached the `userId` and then try to `changeUser` to that same ID, in which case `changeUser` would make no request but `openSession` would, so we'd want to wait on that).